src/structarray.jl

```diff
@@ -134,11 +134,11 @@ Base.axes(s::StructArray) = axes(fieldarrays(s)[1])
 Base.axes(s::StructArray{<:Any, <:Any, <:EmptyTup}) = (1:0,)
 
 get_ith(cols::NamedTuple, I...) = get_ith(Tuple(cols), I...)
-function get_ith(cols::NTuple{N, Any}, I...) where N
-    ntuple(N) do i
```

Comment:
Might need to be 

```
ntuple(Val(N))` do i
   Base.@_inline_meta
  @inbounds res = getfield(cols, i)[I...]
 end
```

But the generated function works as well.

Reply:
I tried

```julia
function get_ith(cols::NTuple{N, Any}, I...) where N
    ntuple(Val(N)) do i
        Base.@_inline_meta
        @inbounds res = getfield(cols, i)[I...]
    end
end
```

but still got a dynamic function

```
ERROR: LoadError: InvalidIRError: compiling kernel!(StructArray{SHermitianCompact{2,Float64,3},1,NamedTuple{(:lowertriangle,),Tuple{StructArray{SArray{Tuple{3},Float64,1,3},1,NamedTuple{(:data,),Tuple{StructArray{Tuple{Float64,Float64,Float64},1,Tuple{CuDeviceArray{Float64,1,CUDAnative.AS.Global},CuDeviceArray{Float64,1,CUDAnative.AS.Global},CuDeviceArray{Float64,1,CUDAnative.AS.Global}},Int64}}},Int64}}},Int64}, StructArray{SHermitianCompact{2,Float64,3},1,NamedTuple{(:lowertriangle,),Tuple{StructArray{SArray{Tuple{3},Float64,1,3},1,NamedTuple{(:data,),Tuple{StructArray{Tuple{Float64,Float64,Float64},1,Tuple{CuDeviceArray{Float64,1,CUDAnative.AS.Global},CuDeviceArray{Float64,1,CUDAnative.AS.Global},CuDeviceArray{Float64,1,CUDAnative.AS.Global}},Int64}}},Int64}}},Int64}) resulted in invalid LLVM IR
Reason: unsupported dynamic function invocation (call to ntuple(f, ::Val{1}) in Base at ntuple.jl:41)
```

I am not sure why.

Reply:
Generally, I would prefer a normal function over a generated function. What is the cause of "dynamic function invocation"? Is it due to the `I...` splatting?

Reply:
I agree it would be nice to avoid generated functions.  I am not sure why this has to be generated but there is definitely something wrong with what I did for `foreachfield` as the test suite doesn't pass.  Maybe that is causing issues.  If you get a chance, could you take a peak at what I did and see if you can spot something astray?